RMET-3643 Firebase Analytics - Add support for Capacitor (MABS 12)

alexgerardojacinto opened the pull request:
## Description
<!--- Describe your changes in detail -->

- Removes dependency to `cordova-outsystems-firebase-core` which is not needed anymore. This plugin wasn't doing anything anymore, as it doesn't have any logic or hooks to run.
- Also adds `build-actions` folder to repo, with information on the build actions used in the plugin.

- Changes on Android
    - Removes `build_gradle_add_dependency.js` hook, which wasn't being used anymore.
    - Changes `config-file` entries in plugin.xml to `edit-config` entries, to achieve the same result in the `AndroidManifest.xml`. This is necessary for MABS 12 apps, as config-file does not work for Capacitor.
    - Also removes the dependency to `cordova-support-android-plugin`, as we cannot include dependencies like this through the `plugin.xml` for Capacitor apps.
   - Because of that, we had to change `FirebaseAnalyticsPlugin.java` to extend `CordovaPlugin` instead of `ReflectiveCordovaPlugin`. This is actually better because this way this plugin is following our practices in other plugins, as we normally don't use this `ReflectiveCordovaPlugin` class.

- Changes on iOS
    - MABS 12 builds Capacitor apps instead of Cordova. In the Xcode project that MABS 12 generates, the SSL Pinning plugin code lives inside a Pod and not the App itself, and so the import of the Generated Header Name to use Swift code inside of Objective-C code is different.
    - For that reason, we're now using a conditional import to support both MABS 12 and MABS <= 11 builds.
    - We also had to mark Swift APIs in different files as public to be able to use them in the Objective-C code.

## Context
<!--- Why is this change required? What problem does it solve? -->
<!--- Place the link to the issue here -->

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

## Type of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply -->
- [ ] Fix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Refactor (cosmetic changes)
- [ ] Breaking change (change that would cause existing functionality to not work as expected)

## Platforms affected
- [x] Android
- [x] iOS
- [ ] JavaScript

## Tests
<!--- Describe how you tested your changes in detail -->
<!--- Include details of your test environment if relevant -->

- Most recent MABS 12 builds of the TestFirebaseAnalytics app on our ODC tenant.

## Screenshots (if appropriate)

## Checklist
<!--- Go over all the following items and put an `x` in all the boxes that apply -->
- [x] Pull request title follows the format `RNMT-XXXX <title>`
- [x] Code follows code style of this project
- [x] CHANGELOG.md file is correctly updated
- [ ] Changes require an update to the documentation
	- [ ] Documentation has been updated accordingly

alexgerardojacinto committed:
chore: remove dependency to `cordova-outsystems-firebase-core`

Context: This Cordova plugin doesn't have any hooks or any logic anymore, so we don't need it.

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
refactor: Remove dependency on cordova-support-android-plugin

By making the native android plugin extend just from plain CordovaPlugin

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
chore: remove unused hook

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: replace `config-file` wth `edit-config` so that it works for Capacitor apps

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: replace `config-file` wth `edit-config` so that it works for Capacitor apps

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
chore: update changelog

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

cssecautomation commented:
![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/pr-banner-default.svg)

### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/native-mobile-xul/pr-checks/102f8c27-86e9-4712-b573-54aa5a08afb0?refs=441f00f6666583dcfc19ab55d11794f7fe226570&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/native-mobile-xul/pr-checks/102f8c27-86e9-4712-b573-54aa5a08afb0/license?refs=441f00f6666583dcfc19ab55d11794f7fe226570&source=prChecksComment)

:white_check_mark: **code/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/native-mobile-xul/project/ff41f515-ce6e-4e21-bd50-34d07bc83f08/pr-check/d6ec68dc-c8bc-49e2-b1f2-b5cfe1d37572?refs=441f00f6666583dcfc19ab55d11794f7fe226570&source=prChecksComment)

alexgerardojacinto committed:
chore: update latest version in changelog

OS-pedrogustavobilro commented on CHANGELOG.md:
```diff
@@ -6,6 +6,16 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+## 5.0.0-OS18
+
+### Features
+- [iOS and Android] Adds support for Capacitor (MABS 12) (https://outsystemsrd.atlassian.net/browse/RMET-3643).
```
Minor: Should we mention MABS 12 here, or maybe just Capacitor?

alexgerardojacinto replied:
Yeah I think you're right, probably just Capacitor

alexgerardojacinto committed:
fix: import Swift code correctly for Capacitor apps

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
chore: update changelog

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: set some properties and functions as public

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: revert previous change

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
test: testing with public types

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
test: remove public keyword

alexgerardojacinto committed:
Revert "test: remove public keyword"

This reverts commit 1f953d5091dad64e6ecb19b23fa122e3e4a334bb.

alexgerardojacinto committed:
chore: add build-actions directory

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto commented:
**Note:** Leaving this PR open until we can use conditional operations on the build actions, because then the YAML file will probably change.

alexgerardojacinto committed:
test: remove edit-config tag

alexgerardojacinto committed:
Revert "test: remove edit-config tag"

This reverts commit 50c52b346c434b987966af7f01c4aca37ec10f7d.

alexgerardojacinto committed:
feat: add ENABLE_APP_TRACKING_TRANSPARENCY_PROMPT to YAML

Context: The ENABLE_APP_TRACKING_TRANSPARENCY_PROMPT setting determines if we should add the NSUserTrackingUsageDescription to the .plist or not.

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
feat: add missing .plist entry for FIREBASE_ANALYTICS_COLLECTION_ENABLED

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: remove 'google_analytics_automatic_screen_reporting_enabled' from YAML

Context: The current version of the plugin for Cordova always sets this preference to 'true' using the plugin.xml. We should keep the same behaviour in this new version, so we should leave it being set to 'true' in the plugin.xml.

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: always set 'FirebaseAutomaticScreenReportingEnabled' to 'true'

Context: The current version of the plugin for Cordova always sets this to true using the plugin.xml. We should keep the same behaviour, so we should set it to true. We need to set it through the YAML and not plugin.xml because for iOS it doesn't work setting it through the plugin.xml for Capacitor apps.

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
feat: add 'google_analytics_automatic_screen_reporting_enabled' in YAML as well

Context: We already add it to the AndroidManifest.xml file of 'capacitor-cordova-android-plugins' using a config-file entry in the plugin.xml, but we might as well add it to the app's AndroidManifest.xml file by setting it in the build action. There won't be a problem of writing the same value twice because it always has the same value (true).

References: https://outsystemsrd.atlassian.net/browse/RMET-3643

alexgerardojacinto committed:
fix: properly include two AndroidManifest.xml entries

References: https://outsystemsrd.atlassian.net/browse/RMET-3643
